### DISCUSSION TITLE
How to change function selection

### DISCUSSION THREAD

**Discussion by DenisSvirin** [MilesCranmer/PySR#241] [Q&A]
How to make hall of fame with  functions with the smallest loss?

By default, i guess, it selects functions with respect to score, as i always have a constant member.

**Comment by MilesCranmer** [comment]
1. You can change the `model_selection` parameter: https://astroautomata.com/PySR/api/#the-objective
2. You can manually select based on your own metric. `predict`, `sympy`, `jax`, etc., all have an `index` parameter. Simply pass the row of the equation you want to use (you can compute this from `model.equations_`, which is the dataframe holding the equations). If `index` is `None` (default), then the `model_selection` strategy is used.

**Reply by DenisSvirin** [reply]
but it doesn't change hall of fame.
It only alters final model output.

**Reply by MilesCranmer** [reply]
Sorry I don't understand. Can you give an example of what you mean?

**Reply by MilesCranmer** [reply]
The hall of fame is the lowest loss at each complexity. "Score" is basically the drop in loss from the previous complexity, but the hall of fame is not selected based on score, only loss.

**Reply by DenisSvirin** [reply]
Oh, sorry. I thought it selects based on score

**Reply by JiaxiangYi96** [reply]
@MilesCranmer  Hello, I am using PySR right now,  and I want to understand how you compute the value of "score" because it is not that clear from both documentation and the paper. Can you give me a bit more detail on how the score numbers are got , For example in the following example.

![image](https://github.com/MilesCranmer/PySR/assets/88609987/a741456b-a038-486e-a455-2a2278c44f19)

**Reply by MilesCranmer** [reply]
@JiaxiangYi96 it is calculated by the change in log(loss) divided the change in complexity with respect to the next simplest expression in the hall of fame.

See scoring section of this notebook: https://github.com/MilesCranmer/PySR/blob/master/examples/pysr_demo.ipynb. (It's also in the paper)
